File: go/finished/p2/p2.go
package main

import "fmt"

func fib(max int) int {
  a, b := 1, 0
  sum := 0
  for a < max {
    if a % 2 == 0 {
        sum += a
    }
    a, b = b,a+b
  }
    return sum
}
func main(){
   fmt.Println(fib(4000000))
}

File: python/finished/p02/p2.py
#!/usr/bin/env python3
def fib(x):
    a, b = 1, 0
    while a < x:
        yield a
        a, b = b, b+a
        print(a)

print(sum(filter(lambda x: x % 2 == 0, fib(4000000))))
